bigtable/scheduled-backups/src/test/java/com/example/cloud/bigtable/scheduledbackups/CreateBackupTest.java

```diff
@@ -0,0 +1,151 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.cloud.bigtable.scheduledbackups;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.api.gax.grpc.GrpcStatusCode;
+import com.google.api.gax.longrunning.OperationFuture;
+import com.google.api.gax.longrunning.OperationFutures;
+import com.google.api.gax.longrunning.OperationSnapshot;
+import com.google.api.gax.rpc.OperationCallable;
+import com.google.api.gax.rpc.UnaryCallable;
+import com.google.api.gax.rpc.testing.FakeOperationSnapshot;
+import com.google.bigtable.admin.v2.CreateBackupMetadata;
+import com.google.cloud.bigtable.admin.v2.BigtableTableAdminClient;
+import com.google.cloud.bigtable.admin.v2.internal.NameUtil;
+import com.google.cloud.bigtable.admin.v2.models.Backup;
+import com.google.cloud.bigtable.admin.v2.models.CreateBackupRequest;
+import com.google.cloud.bigtable.admin.v2.stub.EnhancedBigtableTableAdminStub;
+import com.google.longrunning.Operation;
+import com.google.protobuf.Timestamp;
+import com.google.protobuf.util.Timestamps;
+import io.grpc.Status.Code;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.threeten.bp.Instant;
+
+
+@RunWith(JUnit4.class)
+public final class CreateBackupTest {
+  @Rule public final MockitoRule mockitoRule = MockitoJUnit.rule();
+
+  private static final String PROJECT_ID = "my-project";
+  private static final String INSTANCE_ID = "my-instance";
+  private static final String TABLE_ID = "my-table";
+  private static final String CLUSTER_ID = "my-cluster";
+  private static final String BACKUP_ID = "my-backup";
+
+  private static final String TABLE_NAME =
+      NameUtil.formatTableName(PROJECT_ID, INSTANCE_ID, TABLE_ID);
+
+  private BigtableTableAdminClient adminClient;
+  @Mock private EnhancedBigtableTableAdminStub mockStub;
+
+  @Mock
+  private UnaryCallable<com.google.bigtable.admin.v2.CreateBackupRequest, Operation>
+      mockCreateBackupCallable;
+
+  @Mock
+  private OperationCallable<
+          com.google.bigtable.admin.v2.CreateBackupRequest,
+          com.google.bigtable.admin.v2.Backup,
+          CreateBackupMetadata>
+      mockCreateBackupOperationCallable;
+
+  @Before
+  public void setUp() {
+    adminClient = BigtableTableAdminClient.create(PROJECT_ID, INSTANCE_ID, mockStub);
+
+    Mockito.when(mockStub.createBackupOperationCallable())
+        .thenReturn(mockCreateBackupOperationCallable);
+    Mockito.when(mockStub.createBackupCallable()).thenReturn(mockCreateBackupCallable);
+  }
+
+  @Test
+  public void close() {
+    adminClient.close();
+    Mockito.verify(mockStub).close();
+  }
+
+  @Test
+  public void testCreateBackup() {
```

Comment:
I'm not sure if we need a test like this - this tests the backup API functionality, it would be good to test more of what the example does (ie assert a schedule for a backup was created)

Reply:
Agreed, this is a unit test, we should add an integration test.